framework/guid/model.py

```diff
@@ -49,12 +54,27 @@ def _ensure_guid(self):
             )
             guid.save()
 
-        # Else create GUID optimistically
+        # Else try to create GUID using database of "clean" guids, otherwise default to random generation
         else:
-
-            # Create GUID
-            guid = Guid()
-            guid.save()
+            while True:
+                # Get a clean guid from the database
+                clean_guid = CleanGuid.find()
+                if clean_guid:
+                    rand = random.randint(0, clean_guid.count() - 1)
+                    id = clean_guid[rand]._id
+                else:
+                    id = ''.join(random.sample(ALPHABET, 5))
```

Comment:
Is it necessary to generate a random ID like this? This code wasn't here before, so I'm not sure it's needed.